src/main/java/org/ow2/proactive/catalog/repository/CatalogObjectGrantRepository.java

```diff
@@ -78,4 +78,16 @@
     @Query(value = "SELECT coge.bucketEntity.id FROM CatalogObjectGrantEntity coge WHERE coge.grantee in ?1 AND coge.granteeType='group'")
     List<Long> findAllBucketsIdFromCatalogObjectGrantsAssignedToAUserGroup(List<String> userGroups);
 
+    @Lock(LockModeType.PESSIMISTIC_WRITE)
+    @QueryHints({ @QueryHint(name = "javax.persistence.lock.timeout", value = "5000") })
+    @Query(value = "SELECT coge FROM CatalogObjectGrantEntity coge WHERE coge.grantee = ?1 AND coge.granteeType='user'")
+    List<CatalogObjectGrantEntity> findAllGrantsAssignedToAUsername(String username);
+
+    @Lock(LockModeType.PESSIMISTIC_WRITE)
```

Comment:
I don't think you need exclusive write locks on SELECT statements.
I believe these locks should be on the queries that change the state of the object in the DB.

You are preventing 2 simultaneous read queries on the same table, which isn't an issue right ?

Please get another opinion about this.

Reply:
actually, this seems like a copy paste from other search methods.

I don't know why these lock timeout have been put in other methods in the first place.

Maybe we need to investigate the commit history ?

Reply:
When I wrote them I did similar to other repository in the catalog project (BucketRepository, CatalogObjectRep, ...). So I said maybe they are used to fix a bug that used to occurs or something. I will remove them and test if there is some wrong behaviours.

Reply:
I removed them and tested. Everything seems normal.